src/compile/mark/init.ts

```diff
@@ -98,10 +98,10 @@ function orient(mark: Mark, encoding: Encoding<string>, specifiedOrient: Orienta
 
   switch (mark) {
     case BAR:
-      if (isFieldDef(x) && isBinned(x.bin)) {
+      if (isFieldDef(x) && (isBinned(x.bin) || (isFieldDef(y) && y.aggregate && !x.aggregate))) {
```

Comment:
Should we add the case where there are `x`m `y` and `y2` but not `x2`?

Reply:
Can you give me an example? I don't know how to answer your questions but we should cover all reasonable cases.

Reply:
For example, if there are `x`, `y`, and `y2` in the encoding channel, it should be `"vertical"`
```javascript
{
  mark: "bar",
  encoding: {
    x: {type: "quantitative", field: "field_for_x"},
    y: {type: "quantitative", field: "field_for_y"},
    y2: {field: "field_for_y2"}
  }
}
```

Reply:
Unfortunately, binned data looks the same: https://vega.github.io/vega-lite/docs/bin.html#binned and the orientation should be different.

Reply:
I see.